core/src/main/java/org/apache/iceberg/MetadataUpdate.java

```diff
@@ -543,6 +544,40 @@ public void applyTo(ViewMetadata.Builder viewMetadataBuilder) {
     }
   }
 
+  class AddEncryptionKey implements MetadataUpdate {
```

Comment:
Related to my other comment in https://github.com/apache/iceberg/pull/12927/files#r2068391760 adding an`EncryptionKey` makes much more sense to me and semantically means that we're adding a key that is being used to encrypt table metadata with.